QEMU crashes on WSL with malloc(): invalid size (unsorted) Aborted (core dumped)

Configure command:  ../configure --enable-sdl --disable-cocoa --target-list=arm-softmmu --disable-capstone --disable-slirp --extra-cflags=-I/usr/local/opt/openssl@3/include --extra-ldflags='-L/usr/local/opt/openssl@3/lib -lcrypto' --disable-werror
[log.txt](https://github.com/devos50/qemu/files/10316296/log.txt)

I think this compile warning has the issue:
```
../hw/arm/ipod_touch_adm.c: In function ‘ipod_touch_adm_write’:
../hw/arm/ipod_touch_adm.c:178:32: warning: array subscript ‘uint32_t {aka unsigned int}[0]’ is partly outside array bounds of ‘unsigned char[1]’ [-Warray-bounds]
  178 |                         bank = *buf;
      |                                ^~~~
../hw/arm/ipod_touch_adm.c:176:31: note: object of size 1 allocated by ‘malloc’
  176 |                         buf = malloc(1);
      |                               ^~~~~~~~~
../hw/arm/ipod_touch_adm.c:115:37: warning: array subscript ‘uint32_t {aka unsigned int}[0]’ is partly outside array bounds of ‘unsigned char[2]’ [-Warray-bounds]
  115 |                         num_pages = *buf;
      |                                     ^~~~
../hw/arm/ipod_touch_adm.c:113:31: note: object of size 2 allocated by ‘malloc’
  113 |                         buf = malloc(2);
      |                               ^~~~~~~~~
../hw/arm/ipod_touch_adm.c:157:40: warning: array subscript ‘uint32_t {aka unsigned int}[0]’ is partly outside array bounds of ‘unsigned char[1]’ [-Warray-bounds]
  157 |                                 bank = *buf;
      |                                        ^~~~
../hw/arm/ipod_touch_adm.c:155:39: note: object of size 1 allocated by ‘malloc’
  155 |                                 buf = malloc(1);
      |                                       ^~~~~~~~~
../hw/arm/ipod_touch_adm.c:120:36: warning: array subscript ‘uint32_t {aka unsigned int}[0]’ is partly outside array bounds of ‘unsigned char[2]’ [-Warray-bounds]
  120 |                             bank = *buf;
      |                                    ^~~~
../hw/arm/ipod_touch_adm.c:113:31: note: object of size 2 allocated by ‘malloc’
  113 |                         buf = malloc(2);
      |                               ^~~~~~~~~
../hw/arm/ipod_touch_adm.c:78:37: warning: array subscript ‘uint32_t {aka unsigned int}[0]’ is partly outside array bounds of ‘unsigned char[2]’ [-Warray-bounds]
   78 |                         num_pages = *buf;
      |                                     ^~~~
../hw/arm/ipod_touch_adm.c:76:31: note: object of size 2 allocated by ‘malloc’
   76 |                         buf = malloc(2);
      |                               ^~~~~~~~~
```

--- a/hw/arm/ipod_touch_adm.c
+++ b/hw/arm/ipod_touch_adm.c
@@ -75,7 +75,7 @@ static void ipod_touch_adm_write(void *opaque, hwaddr offset, uint64_t value, un
                         s->nand_state->reading_multiple_pages = true;
                         buf = malloc(2);
                         address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x28, MEMTXATTRS_UNSPECIFIED, buf, 2);
-                        num_pages = *buf;
+                        num_pages = *(uint16_t*)buf;
                         num_pages = ( (((num_pages) >> 8) & 0x00FF) | (((num_pages) << 8) & 0xFF00) ); // swap endianness
                         //printf("Reading %d pages at once, ", num_pages);
 
@@ -112,12 +112,12 @@ static void ipod_touch_adm_write(void *opaque, hwaddr offset, uint64_t value, un
                         s->nand_state->reading_multiple_pages = false;
                         buf = malloc(2);
                         address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x28, MEMTXATTRS_UNSPECIFIED, buf, 2);
-                        num_pages = *buf;
+                        num_pages = *(uint16_t*)buf;
                         num_pages = ( (((num_pages) >> 8) & 0x00FF) | (((num_pages) << 8) & 0xFF00) ); // swap endianness
                         if(num_pages == 1) {
                             // TODO this can probably be refactored to re-use the logic to read multiple pages!
                             address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x44, MEMTXATTRS_UNSPECIFIED, buf, 1);
-                            bank = *buf;
+                            bank = *(uint8_t*)buf;
 
                             buf = malloc(4);
                             address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x244, MEMTXATTRS_UNSPECIFIED, buf, 4);
@@ -154,7 +154,7 @@ static void ipod_touch_adm_write(void *opaque, hwaddr offset, uint64_t value, un
 
                                 buf = malloc(1);
                                 address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x44 + i, MEMTXATTRS_UNSPECIFIED, buf, 1);
-                                bank = *buf;
+                                bank = *(uint8_t*)buf;
                                 // printf("Page: %d, bank: %d\n", page, bank);
 
                                 s->nand_state->pages_to_read[i] = page;
@@ -175,7 +175,7 @@ static void ipod_touch_adm_write(void *opaque, hwaddr offset, uint64_t value, un
                         // writing a page
                         buf = malloc(1);
                         address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x44, MEMTXATTRS_UNSPECIFIED, buf, 1);
-                        bank = *buf;
+                        bank = *(uint8_t*)buf;
 
                         buf = malloc(4);
                         address_space_read(&s->downstream_as, s->data2_sec_addr + 0x1104 + 0x244, MEMTXATTRS_UNSPECIFIED, buf, 4);

--- a/hw/arm/ipod_touch_multitouch.c
+++ b/hw/arm/ipod_touch_multitouch.c
@@ -221,7 +221,7 @@ static uint32_t ipod_touch_multitouch_transfer(SSIPeripheral *dev, uint32_t valu
         uint32_t data_len = (s->in_buffer[2] << 10) | (s->in_buffer[3] << 2) + 5;
         // extend the lengths of the in/out buffers
         free(s->in_buffer);
-        s->in_buffer = malloc(data_len);
+        s->in_buffer = malloc(data_len + 0x10);
 
         free(s->out_buffer);
         s->out_buffer = malloc(data_len);

